model/atmosphere/advection/src/icon4py/model/atmosphere/advection/advection.py

```diff
@@ -0,0 +1,429 @@
+                prep_adv=prep_adv,
+                p_tracer_now=p_tracer_now,
+                p_tracer_new=p_tracer_new,
+                rhodz_now=diagnostic_state.airmass_now,
+                rhodz_new=self._rhodz_ast2,
+                p_mflx_tracer_v=diagnostic_state.vfl_tracer,
+                dtime=dtime,
+                even_timestep=self._even_timestep,
+            )
+
+            # horizontal transport
+            self._horizontal_advection.run(
+                prep_adv=prep_adv,
+                p_tracer_now=p_tracer_new,
+                p_tracer_new=p_tracer_new,
+                rhodz_now=self._rhodz_ast2,
+                rhodz_new=diagnostic_state.airmass_new,
+                p_mflx_tracer_h=diagnostic_state.hfl_tracer,
+                dtime=dtime,
+            )
+
+        else:
+            # horizontal transport
+            self._horizontal_advection.run(
+                prep_adv=prep_adv,
+                p_tracer_now=p_tracer_now,
+                p_tracer_new=p_tracer_new,
+                rhodz_now=diagnostic_state.airmass_now,
+                rhodz_new=self._rhodz_ast2,
+                p_mflx_tracer_h=diagnostic_state.hfl_tracer,
+                dtime=dtime,
+            )
+
+            # vertical transport
+            self._vertical_advection.run(
+                prep_adv=prep_adv,
+                p_tracer_now=p_tracer_new,
+                p_tracer_new=p_tracer_new,
+                rhodz_now=self._rhodz_ast2,
+                rhodz_new=diagnostic_state.airmass_new,
+                p_mflx_tracer_v=diagnostic_state.vfl_tracer,
+                dtime=dtime,
+                even_timestep=self._even_timestep,
+            )
+
+        # update lateral boundaries with interpolated time tendencies
+        if self._grid.limited_area:
+            log.debug("running stencil apply_interpolated_tracer_time_tendency - start")
+            apply_interpolated_tracer_time_tendency.apply_interpolated_tracer_time_tendency(
+                p_tracer_now=p_tracer_now,
+                p_grf_tend_tracer=diagnostic_state.grf_tend_tracer,
+                p_tracer_new=p_tracer_new,
+                p_dtime=dtime,
+                horizontal_start=self._start_cell_lateral_boundary,
+                horizontal_end=self._end_cell_lateral_boundary_level_4,
+                vertical_start=0,
+                vertical_end=self._grid.num_levels,
+                offset_provider=self._grid.offset_providers,
+            )
+            log.debug("running stencil apply_interpolated_tracer_time_tendency - end")
+
+        # exchange updated tracer values, originally happens only if iforcing /= inwp
+        log.debug("communication of advection cell field: p_tracer_new - start")
+        self._exchange.exchange_and_wait(dims.CellDim, p_tracer_new)
+        log.debug("communication of advection cell field: p_tracer_new - end")
+
+        # finalize step
+        self._even_timestep = not self._even_timestep
```

Comment:
shouldn't this be called 
```suggestion
        self._odd_timestep = not self._even_timestep
```
since it's the opposite of even?